src/gitchangelog/gitchangelog.py

```diff
@@ -1349,6 +1355,102 @@ def render_commit(commit, opts=opts):
+    """Returns Markdown Text changelog content from data
+
+    Connect to GitHub if commit is missing a body and retrieve PR description.
+    """
+
+    token = opts.get("github_token")
+    repo = opts.get("github_repo")
+    if not all([token, repo]):
+        die("Github token, repo and/or PR regexp config is missing.")
+
+    # GitHub
+    github = None
+    RE_PR_NUM = None
+    if Github:
+        try:
+            github = Github(token)
+            repo = github.get_repo(repo)
+        except Exception:
+            die("Unable to connect to Github")
+
+        RE_PR_NUM = re.compile(
+            # Example: "Title of commit (#PR_NUM)"
+            r"""
+                \(
+                \#(\d+)     # wrapped in parentheses for direct number match
+                \)
+                $
+            """,
+            re.X,
+        )
+        # Extract only description section from Github PR body
+        RE_PR_DESC = re.compile(
+            # Example: PR template markdown file in git repo
+            r"##[ ]*Description(?P<desc>.*)##[ ]*Preflight Checklist",
+            re.DOTALL,
+        )
+
+    def rest_title(label, char="="):
+        return (label.strip() + "\n") + (char * len(label) + "\n")
+
+    def render_version(version):
+        title = "%s (%s)" % (version["tag"], version["date"]) \
+                if version["tag"] else \
+                opts["unreleased_version_label"]
+        s = rest_title(title, char="-")
+
+        sections = version["sections"]
+        nb_sections = len(sections)
+        for section in sections:
+
+            section_label = section["label"] if section.get("label", None) \
+                            else "Other"
+
+            if not (section_label == "Other" and nb_sections == 1):
+                s += "\n" + rest_title(section_label, "~")
+
+            for commit in section["commits"]:
+                s += render_commit(commit)
+        return s
+
+    def render_commit(commit, opts=opts):
+        subject = commit["subject"]
+
+        entry = indent('\n'.join(textwrap.wrap(subject)),
+                       first="- ").strip() + "\n"
+
+        if commit["body"]:
+            entry += "\n" + indent(commit["body"]) + "\n"
+        else:
+            if RE_PR_NUM:
+                # Get GitHub PR description/body
+                pr_num = RE_PR_NUM.search(commit["subject"])
+                pr_num = pr_num.groups()[0] if pr_num else None
+                if pr_num:
+                    try:
+                        pull = repo.get_pull(int(pr_num))
+                        body = RE_PR_DESC.search(pull.body)
+                        body = body.groupdict()['desc'].strip() if body else ""
+                        if body:
+                            entry += "\n```\n" + body + "\n```\n"
+                    except Exception as e:
+                        err(f"Unable to retrieve PR #{pr_num} from Github.")
+                        err(f"Exception: {e}")
```

Comment:
I get syntax error on my Python setup which is kind of weird. This construction works though:
```
err("Unable to retrieve PR #%s from Github." % (pr_num, ))
err("Exception: %s" % (e, ))
```

Reply:
@lookashc That syntax works on Python 3.6 and later.

Reply:
I had issues even on python 3.8.5 (we were checking it with @drazenzen) but maybe my setup was incorrect. Glad to have it working now 💯